pkg/operation/cloudbotanist/packetbotanist/infrastructure.go

```diff
@@ -0,0 +1,93 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package packetbotanist
+
+import (
+	"github.com/gardener/gardener/pkg/operation/common"
+	"github.com/gardener/gardener/pkg/operation/terraformer"
+	"github.com/gardener/gardener/pkg/utils/flow"
+)
+
+const (
+	projectID = "projectID"
+	sshKey = "sshKey"
+)
+
+// DeployInfrastructure kicks off a Terraform job which deploys the infrastructure.
+func (b *PacketBotanist) DeployInfrastructure() error {
+	tf, err := b.NewShootTerraformer(common.TerraformerPurposeInfra)
+	if err != nil {
+		return err
+	}
+	return tf.
+		SetVariablesEnvironment(b.generateTerraformInfraVariablesEnvironment()).
+		InitializeWith(b.ChartInitializer("packet-infra", b.generateTerraformInfraConfig())).
+		Apply()
+}
+
+// DestroyInfrastructure kicks off a Terraform job which destroys the infrastructure.
+func (b *PacketBotanist) DestroyInfrastructure() error {
+        tf, err := b.NewShootTerraformer(common.TerraformerPurposeInfra)
+        if err != nil {
+                return err
+        }
+
+        var (
+                g = flow.NewGraph("Packet infrastructure destruction")
+
+                _ = g.Add(flow.Task{
+                        Name:         "Destroying Shoot infrastructure",
+                        Fn:           flow.SimpleTaskFn(tf.SetVariablesEnvironment(b.generateTerraformInfraVariablesEnvironment()).Destroy),
+                })
+
+                f = g.Compile()
+        )
+
+        if err := f.Run(flow.Opts{Logger: b.Logger}); err != nil {
+                return flow.Causes(err)
+        }
+        return nil
+}
+
+// generateTerraformInfraVariablesEnvironment generates the environment containing the credentials which
+// are required to validate/apply/destroy the Terraform configuration. These environment must contain
+// Terraform variables which are prefixed with TF_VAR_.
+func (b *PacketBotanist) generateTerraformInfraVariablesEnvironment() map[string]string {
+	return terraformer.GenerateVariablesEnvironment(b.Shoot.Secret, map[string]string{
+		"PACKET_API_KEY":     PacketAPIKey,
+	})
+}
+
+// generateTerraformInfraConfig creates the Terraform variables and the Terraform config (for the infrastructure)
+// and returns them (these values will be stored as a ConfigMap and a Secret in the Garden cluster.
+func (b *PacketBotanist) generateTerraformInfraConfig() map[string]interface{} {
+	return map[string]interface{}{
+		"projectID": string(b.Shoot.Secret.Data[projectID]),
+		"sshPublicKey": string(b.Shoot.Secret.Data[sshKey]),
+		"clusterName": b.Shoot.SeedNamespace,
+	}
+}
+
+// DeployBackupInfrastructure kicks off a Terraform job which deploys the infrastructure resources for backup.
```

Comment:
As discussed previously, maybe let's write a comment here that it is planned to use what is described in #928 as Packet itself does not have a blob store?

Reply:
Sure, easy enough to add.